File: src/main/java/com/icloud/state/before/Student.java
package com.icloud.state.before;

import java.util.ArrayList;
import java.util.List;

public class Student {

    private final String name;

    public Student(String name) {
        this.name = name;
    }

    private List<OnlineCourse> privateOnlineCourses = new ArrayList<>();

    public void addPrivateCourse(OnlineCourse onlineCourse) {
        this.privateOnlineCourses.add(onlineCourse);
    }

    public boolean isEnabledForPrivateClass(OnlineCourse onlineCourse) {
        return this.privateOnlineCourses.contains(onlineCourse);
    }

    @Override
    public String toString() {
        return "Student{" +
                "name='" + name + '\'' +
                ", privateOnlineCourses=" + privateOnlineCourses +
                '}';
    }
}

File: src/main/java/com/icloud/state/review/Student.kt
package com.icloud.state.review

class Student(var name: String) {

    var onlineCourses: MutableSet<OnlineCourse> = mutableSetOf<OnlineCourse>()

    fun isAvailable(onlineCourse: OnlineCourse): Boolean {
        return this.onlineCourses.contains(onlineCourse)
    }

    fun addPrivate(onlineCourse: OnlineCourse) {
        this.onlineCourses.add(onlineCourse)
    }

    override fun toString(): String {
        return "Student(name='$name')"
    }


}